--- octave-3.0.3/liboctave/sparse-util.h~	2008-09-24 09:13:47.000000000 +0200
+++ octave-3.0.3/liboctave/sparse-util.h	2009-02-26 01:57:30.000000000 +0100
@@ -24,7 +24,7 @@
 #if !defined (octave_sparse_util_h)
 #define octave_sparse_util_h 1
 
-extern OCTAVE_API void SparseCholError (int status, char *file, int line, char *message);
+extern OCTAVE_API void SparseCholError (int status, const char *file, int line, const char *message);
 extern OCTAVE_API int SparseCholPrint (const char *fmt, ...);
 
 #endif
--- octave-3.0.3/liboctave/sparse-util.cc~	2008-09-24 09:13:47.000000000 +0200
+++ octave-3.0.3/liboctave/sparse-util.cc	2009-02-26 01:57:08.000000000 +0100
@@ -31,7 +31,7 @@
 #include "sparse-util.h"
 
 void
-SparseCholError (int status, char *file, int line, char *message)
+SparseCholError (int status, const char *file, int line, const char *message)
 {
   (*current_liboctave_warning_handler)("warning %i, at line %i in file %s",
 				     status, line, file);
