--- octave-3.6.4/src/mex.h~	2013-02-21 21:19:24.000000000 +0100
+++ octave-3.6.4/src/mex.h	2013-06-24 13:08:46.755012288 +0200
@@ -64,6 +64,8 @@
 
 #define mxMAXNAME 64
 
+#include "mexproto.h"
+
 #if defined (__cplusplus)
 extern "C" {
 #endif
@@ -74,8 +76,6 @@
 void mexFunction (int nlhs, mxArray* plhs[], int nrhs, const mxArray *prhs[]);
 #endif
 
-#include "mexproto.h"
-
 /* V4 floating point routines renamed in V5.  */
 #define mexIsNaN mxIsNaN
 #define mexIsFinite mxIsFinite
