--- octave-3.2.4/src/DLD-FUNCTIONS/__magick_read__.cc	2009-08-25 10:26:01.000000000 +0200
+++ octave-3.3.51/src/DLD-FUNCTIONS/__magick_read__.cc	2010-03-03 18:49:00.000000000 +0100
@@ -27,6 +28,7 @@
 
 #include <cmath>
 
+#include "oct-env.h"
 #include "defun-dld.h"
 #include "error.h"
 #include "ov-struct.h"
@@ -358,6 +384,17 @@
 
 #ifdef HAVE_MAGICK
 
+  static bool initialized = false;
+
+  if (! initialized)
+    {
+      std::string program_name = octave_env::get_program_invocation_name ();
+
+      Magick::InitializeMagick (program_name.c_str ());
+
+      initialized = true;
+    }
+
   if (args.length () > 2 || args.length () < 1 || ! args(0).is_string ()
       || nargout > 3)
     {
