--- Makeconf.in.orig	Mon Jul 23 16:57:13 2001
+++ Makeconf.in	Mon Jul 23 16:57:16 2001
@@ -160,20 +160,20 @@
 # subdirectories of this directory.  The default values for many of
 # the variables below are expressed in terms of this one, so you may
 # not need to change them.  This defaults to /usr/local.
-prefix = @prefix@
+prefix = $(DESTDIR)@prefix@
 
 # Like `prefix', but used for architecture-specific files.
-exec_prefix = @exec_prefix@
+exec_prefix = $(DESTDIR)@exec_prefix@
 
 # Where to install Octave and other binaries that people will want to
 # run directly.
-bindir = @bindir@
+bindir = $(DESTDIR)@bindir@
 
 # Where to install architecture-independent data files.  ${fcnfiledir}
 # and ${localfcnfiledir} are subdirectories of this.
-datadir = @datadir@
+datadir = $(DESTDIR)@datadir@
 
-libdir = @libdir@
+libdir = $(DESTDIR)@libdir@
 
 # Where to install and expect libraries like libcruft.a, liboctave.a,
 # and other architecture-dependent data.
@@ -181,20 +181,20 @@
 
 # Where to install and expect executable programs to be run by Octave
 # rather than directly by users.
-libexecdir = @libexecdir@
+libexecdir = $(DESTDIR)@libexecdir@
 
 # The prefix for Octave's include file directory.  The default is
 # ${prefix}/include
-includedir = @includedir@
+includedir = $(DESTDIR)@includedir@
 
 # Where to install Octave's man pages, and what extension they should
 # have.  The default is ${prefix}/man/man1
-mandir = @mandir@
+mandir = $(DESTDIR)@mandir@
 man1dir = @man1dir@
 man1ext = @man1ext@
 
 # Where to install and expect the info files describing Octave..
-infodir = @infodir@
+infodir = $(DESTDIR)@infodir@
 
 # The fill path to the default info file.
 infofile = @infofile@
