--- octave-2.9.17/configure.in~	2007-11-10 06:17:27.000000000 +0100
+++ octave-2.9.17/configure.in	2007-12-03 00:20:36.000000000 +0100
@@ -1536,21 +1536,15 @@
 OCTAVE_MKDIR_TAKES_ONE_ARG
 
 octave_found_termlib=no
-for termlib in ncurses curses termcap terminfo termlib; do
-  AC_CHECK_LIB(${termlib}, tputs, [TERMLIBS="${TERMLIBS} -l${termlib}"])
-  case "${TERMLIBS}" in
-    *-l${termlib}*)
-      LIBS="$TERMLIBS $LIBS"
-      AC_SUBST(TERMLIBS)
-      octave_found_termlib=yes
-      break
-    ;;
-  esac
-done
+AC_CHECK_LIB(ncurses, tputs, [TERMLIBS="-lncurses -ltinfo"],,"-ltinfo")
+if test "x$TERMLIBS" != "x"; then
+	LIBS="$TERMLIBS $LIBS"
+	AC_SUBST(TERMLIBS)
+	octave_found_termlib=yes
+fi
 
 if test "$octave_found_termlib" = no; then
-  warn_termlibs="I couldn't find -ltermcap, -lterminfo, -lncurses, -lcurses, o\
-r -ltermlib!"
+  warn_termlibs="I couldn't find -lncurses!"
   AC_MSG_WARN($warn_termlibs)
 fi
 
