--- octave-3.8.1/m4/acinclude.m4.orig	2014-03-06 20:37:51.000000000 +0100
+++ octave-3.8.1/m4/acinclude.m4	2014-03-10 08:01:49.629358095 +0100
@@ -972,7 +972,7 @@
   TERM_LIBS=
   ac_octave_save_LIBS="$LIBS"
   AC_SEARCH_LIBS([tputs],
-                 [ncurses curses termcap terminfo termlib],
+                 [tinfo ncurses curses termcap terminfo termlib],
                  [], [])
   LIBS="$ac_octave_save_LIBS"
   case "$ac_cv_search_tputs" in
--- octave-4.4.1/m4/ax_pthread.m4.orig	2022-12-13 16:59:28.522309474 +0100
+++ octave-4.4.1/m4/ax_pthread.m4	2022-12-13 17:03:38.324289514 +0100
@@ -189,6 +189,7 @@ for flag in $ax_pthread_flags; do
                 -*)
                 AC_MSG_CHECKING([whether pthreads work with $flag])
                 PTHREAD_CFLAGS="$flag"
+                PTHREAD_LIBS="$flag"
                 ;;
 
                 pthread-config)
