--- octave-2.1.48/src/ov-cx-mat.h.orig	2002-11-20 20:03:08.000000000 +0100
+++ octave-2.1.48/src/ov-cx-mat.h	2003-05-17 20:34:16.000000000 +0200
@@ -78,8 +78,7 @@
 
   octave_value *try_narrowing_conversion (void);
 
-  void assign (const octave_value_list& idx, const ComplexMatrix& rhs)
-    { octave_base_matrix<ComplexMatrix>::assign (idx, rhs); }
+  void assign (const octave_value_list& idx, const ComplexMatrix& rhs);
 
   void assign (const octave_value_list& idx, const Matrix& rhs);
 
--- octave-2.1.48/src/ov-cx-mat.cc.orig	2002-11-20 20:03:08.000000000 +0100
+++ octave-2.1.48/src/ov-cx-mat.cc	2003-05-17 20:35:56.000000000 +0200
@@ -78,6 +78,11 @@
 
 void
 octave_complex_matrix::assign (const octave_value_list& idx,
+			       const ComplexMatrix& rhs)
+{ octave_base_matrix<ComplexMatrix>::assign (idx, rhs); }
+
+void
+octave_complex_matrix::assign (const octave_value_list& idx,
 			       const Matrix& rhs)
 {
   int len = idx.length ();
