--- octave-4.4.1/configure.ac.orig	2022-12-13 19:12:53.565609102 +0100
+++ octave-4.4.1/configure.ac	2022-12-13 19:58:22.644157748 +0100
@@ -91,7 +91,7 @@ AC_SUBST(config_opts)
 
 ## Where to install libraries like liboctave.a, liboctinterp.a, and other
 ## architecture-dependent binaries.
-OCTAVE_SET_DEFAULT([octlibdir], '${libdir}/octave/${version}')
+OCTAVE_SET_DEFAULT([octlibdir], '${libdir}')
 
 ## Where to put executables to be run by Octave, rather than the user.
 ## This path usually includes the Octave version and configuration name, so
@@ -130,7 +130,7 @@ OCTAVE_SET_DEFAULT([localveroctfiledir],
   '${libdir}/octave/${version}/site/oct/${canonical_host_type}')
 
 ## Where to install Octave's include files.
-OCTAVE_SET_DEFAULT([octincludedir], '${includedir}/octave-${version}/octave')
+OCTAVE_SET_DEFAULT([octincludedir], '${includedir}/octave')
 
 ## Where to install the function files distributed with Octave.
 ## This includes the Octave version, so that the function files for different
